### ISSUE TITLE
Integration test fail due to certificate problem.

### ISSUE THREAD

**Issue by samsalisbury** [opentable/sous#260]
I just tried running `./bin/dev-integration` on a fresh pull of 50edd4b8bee6ad56a25d8cc8e15a0f37c2465a1e and got the following error: 

    Get https://192.168.99.100:5000/v1/_ping: x509: certificate has expired or is not yet valid

Longer log extract for context:

```
Attempting connection: 192.168.99.100:5000
Attempting connection: 192.168.99.100:7099
  192.168.99.100:5000 up and running
  192.168.99.100:7099 up and running
2016/12/16 10:48:49 docker_helpers.go:71:    *** Beginning tests... ***

2016/12/16 10:48:49 docker_helpers.go:145: Problem pushing container: hello-labels
The push refers to a repository [192.168.99.100:5000/hello-labels]
Get https://192.168.99.100:5000/v1/_ping: x509: certificate has expired or is not yet valid
--- FAIL: TestGetLabels (0.30s)
panic: building test container failed: exit status 1 [recovered]
        panic: building test container failed: exit status 1

goroutine 6 [running]:
panic(0x46ce6e0, 0xc4203c52e0)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
testing.tRunner.func1(0xc420090600)
        /usr/local/go/src/testing/testing.go:579 +0x25d
panic(0x46ce6e0, 0xc4203c52e0)
        /usr/local/go/src/runtime/panic.go:458 +0x243
github.com/opentable/sous/integration.registerAndDeploy(0xc4203c5060, 0x10, 0x10, 0x476440b, 0xc, 0x4763acf, 0xc, 0x4799f91, 0x25, 0x4763acf, ...)
        /Users/ssalisbury/go/src/github.com/opentable/sous/integration/docker_helpers.go:103 +0x2b4
github.com/opentable/sous/integration.registerLabelledContainers()
        /Users/ssalisbury/go/src/github.com/opentable/sous/integration/integration_test.go:393 +0xca
github.com/opentable/sous/integration.TestGetLabels(0xc420090600)
        /Users/ssalisbury/go/src/github.com/opentable/sous/integration/integration_test.go:25 +0x34
testing.tRunner(0xc420090600, 0x47e9070)
        /usr/local/go/src/testing/testing.go:610 +0x81
created by testing.(*T).Run
        /usr/local/go/src/testing/testing.go:646 +0x2ec
FAIL    github.com/opentable/sous/integration   0.321s
```

**Comment by samsalisbury** [comment]
Just tried a forced git clean in case I had some old files hanging around, and got the same problem.

**Comment by samsalisbury** [comment]
This mysteriously started working again for a short time, on both 'git cleaned' and dirty repo. However, it still fails with the above issue sporadically. I have tried opening new shells to make sure it's not caused by a dirty environment, and it still fails sporadically with this issue, and other times with actual test failures:

```
--- FAIL: TestResolve (4.13s)                                                                                                                         
        Error Trace:    integration_test.go:277                                                                                                       
        Error:          Should not be: sous.DeployID{ManifestID:sous.ManifestID{Source:sous.SourceLocation{Repo:"", Dir:""}, Flavor:""}, Cluster:""}  
        Messages:       opentable/one not successfully deployed                                                                                       
                                                                                                                                                      
        Error Trace:    integration_test.go:283                                                                                                       
        Error:          Should not be: sous.DeployID{ManifestID:sous.ManifestID{Source:sous.SourceLocation{Repo:"", Dir:""}, Flavor:""}, Cluster:""}  
        Messages:       opentable/two not successfully deployed                                                                                                                                                                                                                                           
```

**Comment by samsalisbury** [comment]
Got it. This failure is caused by differences in environment. In those where docker-machine environment is loaded, everything works as expected. In others, we get the certificate failure. (The other failure is due to race condition in integration tests themselves, so now it's de-scoped from this issue.)

I think the correct fix here is to assert that we have env loaded for an active docker-machine before trying to set up the qa environment and run the tests, if running on a machine with docker-machine installed.

The docker-machine env vars are:

```
DOCKER_HOST
DOCKER_MACHINE_NAME
DOCKER_TLS_VERIFY
DOCKER_CERT_PATH
```

**Comment by samsalisbury** [comment]
So, this was all caused by a confusing error message when no docker-machine environment was loaded, and no local docker daemon was available. That confusion is solved by #261.

**Comment by samsalisbury** [comment]
Closing as #261 is merged.
